synapse/storage/registration.py

```diff
@@ -254,8 +254,8 @@ def user_delete_access_tokens(self, user_id, except_token_id=None,
                 If None, tokens associated with any device (or no device) will
                 be deleted
         Returns:
-            defer.Deferred[list[str, str|None]]: a list of the deleted tokens
-                and device IDs
+            defer.Deferred[list[str, int, str|None, int]]: a list of
+                (token, token id, device id) for each of the deleted tokens
```

Comment:
Why are we changing this? It doesn't seem to be used elsewhere in the code?

Reply:
we need the token_id so that we can delete the pushers which correspond to the deleted access tokens: https://github.com/matrix-org/synapse/pull/2722/commits/2c6d63922a5033a17c7d53a928892fbbcbd6fa63#diff-b8464485d36f6f87caee3f4d82524213R743